src/find-options/FindOptionsUtils.ts

```diff
@@ -267,7 +268,44 @@ export class FindOptionsUtils {
             let relationAlias = DriverUtils.buildAlias(qb.connection.driver, { shorten: true }, qb.connection.namingStrategy.eagerJoinRelationAlias(alias, relation.propertyPath));
 
             // add a join for the relation
-            qb.leftJoinAndSelect(alias + "." + relation.propertyPath, relationAlias);
+            // Checking whether the relation wasn't joined yet.
+            let addJoin = true;
+            for (const join of qb.expressionMap.joinAttributes) {
+                if (
+                    join.condition !== "" ||
+                    join.mapToProperty !== undefined ||
+                    join.isMappingMany !== undefined ||
+                    join.direction !== "LEFT" ||
+                    join.entityOrProperty !== `${alias}.${relation.propertyPath}`
+                ) {
+                    continue;
+                }
+                addJoin = false;
+                relationAlias = join.alias.name;
+                break;
+            }
+            if (qb.connection.driver.maxAliasLength && relationAlias.length > qb.connection.driver.maxAliasLength) {
+                relationAlias = shorten(relationAlias);
```

Comment:
Isn't this already shortened because of `DriverUtils.buildAlias`?

Reply:
you are right, isn't needed anymore. I removed it.